### PR TITLE
Implement CRUD tests for compute resources for API

### LINKED ISSUE #3258: Add CRUD tests for Compute resource for API
Looks like we have no tests for Compute resource via API (except docker tests), even the `api/test_computeresource.py` file is missing.

### PR DESCRIPTION
**Pull request by abalakh**
Closes #3258 
```python
py.test tests/foreman/api/test_computeresource.py -n auto
================================ test session starts =================================
platform linux2 -- Python 2.7.10, pytest-2.8.7, py-1.4.31, pluggy-0.3.1
rootdir: /home/andrii/workspace/robottelo, inifile: 
plugins: xdist-1.13.1, repeat-0.2
gw0 [20] / gw1 [20] / gw2 [20] / gw3 [20]
scheduling tests via LoadScheduling
....................
============================ 20 passed in 102.06 seconds =============================
```

### CODE REVIEW

File: docs/api/tests.foreman.api.rst [hunk 1]
```diff
@@ -13,6 +13,11 @@
 
 .. automodule:: tests.foreman.api.test_architecture
 
+:mod:`tests.foreman.api.test_computeresource`
+---------------------------------------------
+
+.. automodule:: tests.foreman.api.test_computeresource
```
Comment (elyezer) [inline docs/api/tests.foreman.api.rst]: :+1:

**Comment by elyezer** [comment]
LGTM, ACK.

File: tests/foreman/api/test_computeresource.py [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+from robottelo.constants import LIBVIRT_RESOURCE_URL
+from robottelo.datafactory import (
+    invalid_names_list,
+    invalid_values_list,
+    valid_data_list,
+)
+from robottelo.decorators import tier1, tier2
+from robottelo.test import APITestCase
+
+LIBVIRT_URL = LIBVIRT_RESOURCE_URL.format(settings.server.hostname)
```
Comment (oshtaier) [inline tests/foreman/api/test_computeresource.py]: Please, fix UI and CLI to use your approach with that constant

**Comment by oshtaier** [comment]
If you don't mind, please add test cases for `display_type` parameter

**Comment by oshtaier** [comment]
ACK pending comments

**Comment by abalakh** [comment]
Comments addressed. Because of settings limitations, my approach with constant was wrong, so i've updated CLI and API with UI approach.
As a bonus updated UI compresource tests to work with sat 6.2
Tests results (for api `display_type` and 1 per other interfaces)
```python
py.test tests/foreman/api/test_computeresource.py -k 'display_type' -n auto
================================ test session starts =================================
platform linux2 -- Python 2.7.10, pytest-2.8.7, py-1.4.31, pluggy-0.3.1
rootdir: /home/andrii/workspace/robottelo, inifile: 
plugins: xdist-1.13.1, repeat-0.2
gw0 [2] / gw1 [2] / gw2 [2] / gw3 [2]
scheduling tests via LoadScheduling
..
============================= 2 passed in 12.54 seconds ==============================
py.test tests/foreman/cli/test_computeresource.py -k 'test_positive_create_with_name' 
================================ test session starts =================================
platform linux2 -- Python 2.7.10, pytest-2.8.7, py-1.4.31, pluggy-0.3.1
rootdir: /home/andrii/workspace/robottelo, inifile: 
plugins: xdist-1.13.1, repeat-0.2
collected 14 items 

tests/foreman/cli/test_computeresource.py .

============= 13 tests deselected by '-ktest_positive_create_with_name' ==============
====================== 1 passed, 13 deselected in 7.18 seconds =======================
py.test tests/foreman/ui/test_computeresource.py -k 'test_positive_create_libvirt_with_name'
================================ test session starts =================================
platform linux2 -- Python 2.7.10, pytest-2.8.7, py-1.4.31, pluggy-0.3.1
rootdir: /home/andrii/workspace/robottelo, inifile: 
plugins: xdist-1.13.1, repeat-0.2
collected 10 items 

tests/foreman/ui/test_computeresource.py .

========== 9 tests deselected by '-ktest_positive_create_libvirt_with_name' ==========
====================== 1 passed, 9 deselected in 112.62 seconds ======================
```

**Comment by oshtaier** [comment]
APT pending travis failure

**Comment by oshtaier** [comment]
ACK

**Pull request merged**